jupyter_client/kernelspec.py

```diff
@@ -242,10 +273,12 @@ def install_kernel_spec(self, source_dir, kernel_name=None, user=False,
         if not kernel_name:
             kernel_name = os.path.basename(source_dir)
         kernel_name = kernel_name.lower()
-        
+        if not _valid_kernel_name(kernel_name):
+            raise ValueError("Invalid kernel name %r.  %s" % (kernel_name, _kernel_name_description))
```

Comment:
Should we raise already or have a grace period ? I have kernel spec on my machine that are invalid.